packages/connect/src/createScClient/createScClient.ts

```diff
@@ -11,19 +20,293 @@ export interface ScClient {
+    // you would expect this to be `this.#subscriptions.size > 0`, right?
+    // well, nope... This must always return `true` b/c otherwise Polkadot
+    // thinks that this Provider doesn't support subscriptions.
+    return true
+  }
+
+  public get isConnected(): boolean {
+    return !!this.#chain && this.#isChainReady
+  }
+
+  public clone(): ProviderInterface {
+    throw new Error("clone() is not supported.")
+  }
+
+  async connect(): Promise<void> {
+    if (this.isConnected) throw new Error("Already connected!")
+
+    // it could happen that after emitting `disconnected` due to the fact taht
+    // smoldot is syncing, the consumer tries to reconnect after a certain amount
+    // of time... In which case we want to make sure that we don't create a new
+    // chain.
+    if (this.#chain) {
+      await this.#chain
+      return
+    }
+
+    const hc = healthChecker()
+    const onResponse = (res: string): void => {
+      if (!hc.responsePassThrough(res)) {
+        return
+      }
+
+      const response = JSON.parse(res) as JsonRpcResponse
+
+      // the healthChecker alters the original id. So, we must restore it
+      // because otherwise the RpcCoder won't be able to decode the resonse.
+      if (
+        typeof response.id === "string" &&
+        (response.id as unknown as string).startsWith("extern:")
+      ) {
+        response.id = Number((response.id as unknown as string).slice(7))
+      }
+
+      let decodedResponse: string | Error
+      try {
+        decodedResponse = this.#coder.decodeResponse(response) as string
+      } catch (e) {
+        decodedResponse = e as Error
+      }
+
+      // It's not a subscription message, but rather a standar RPC response
+      if (response.method === undefined) {
+        return this.#requests.get(response.id)?.(decodedResponse)
+      }
+
+      // It has a `method` property, that means that it's a subscription message
+      const subscriptionId = `${response.method}::${response.params.subscription}`
+
+      const callback = this.#subscriptions.get(subscriptionId)?.[0]
+      if (callback) return callback(decodedResponse)
+
+      // It's possible to receive subscriptions messages before having received
+      // the id of the subscription. In that case We should keep these
+      // messages around until we receive the subscription-id message.
+      if (!this.#orphanMessages.has(subscriptionId))
+        this.#orphanMessages.set(subscriptionId, [])
+      this.#orphanMessages.get(subscriptionId)!.push(decodedResponse)
+    }
+
+    this.#chain = this.#getChain(onResponse).then((chain) => {
+      hc.setSendJsonRpc(chain.sendJsonRpc)
+
+      this.#isChainReady = false
+      const cleanup = () => {
+        // If there are any callbacks left, we have to reject/error them.
+        // Otherwise, that would cause a memory leak.
+        const disconnectionError = new Error("Disconnected")
+        this.#requests.forEach((cb) => cb(disconnectionError))
```

Comment:
You must do `this.#requests.clear()`
The reason why we track requests is so that if PolkadotJS disconnects then reconnects we don't want it to receive responses to requests that it has sent before the disconnection.